src/parser/Lexer.cpp

```diff
@@ -620,7 +609,7 @@ void Scanner::ScannerResult::constructStringLiteral()
     } else {
         newStr = new UTF16String(stringUTF16.data(), stringUTF16.length());
     }
-    this->valueStringLiteralData = StringView(newStr, 0, newStr->length());
+    this->valueStringLiteralData = new StringView(newStr, 0, newStr->length());
```

Comment:
before return, check nullity of ``new`` with ASSERT

Reply:
Same as above.